test/evp_test.c

```diff
@@ -337,21 +335,159 @@ typedef struct digest_data_st {
     int pad_type;
 } DIGEST_DATA;
 
+
+/* Done as macros to avoid compiler warnings about unused functions  */
+#define STR_STARTS_WITH(str, pre) strncmp(pre, str, strlen(pre)) == 0
```

Comment:
I don't understand the comment.  And shouldn't that by strncasecmp?

Reply:
If I do this as a function then I need to do #if defined(X) || defined{Y} || define(Z) || .......
otherwise it will produce a error that the function is not used in the case where a user turns everything off with no- options.

Reply:
Still don't get it, but that's okay.

but strncasecmp?

Reply:
Will do..

If you search for STR_STARTS_WITH() you will see each one is inside a #ifdef e.g:

```
#ifdef OPENSSL_NO_POLY1305
    if (STR_STARTS_WITH(name, "Poly1305"))
        return 1;
#endif
```
if STR_STARTS_WITH is a function instead of a macro then the function would be unused if all the OPENSSL_NO_POLY1305 lines are undefined.

Reply:
So the function would need to be wrapped with a nasty #ifdef to pick up this case.

Reply:
or a dummy case provided.  shrug.  doesn't matter to me.